src/Datadog.Trace/AppSec/Waf/NativeBindings/Native.cs

```diff
@@ -39,43 +57,200 @@ internal class Native
         internal static extern IntPtr pw_initAdditiveH(IntPtr powerwafHandle);
 
-        [DllImport("Datadog.Trace.ClrProfiler.Native")]
+        [DllImport(DllName)]
         internal static extern PWRet pw_runAdditive(IntPtr context, PWArgs newArgs, ulong timeLeftInUs);
 
         [DllImport(DllName)]
         internal static extern void pw_clearAdditive(IntPtr context);
 
-        [DllImport("Datadog.Trace.ClrProfiler.Native")]
+        [DllImport(DllName)]
         internal static extern PWArgs pw_getInvalid();
 
-        [DllImport("Datadog.Trace.ClrProfiler.Native")]
+        [DllImport(DllName)]
         internal static extern PWArgs pw_createStringWithLength(string s, ulong length);
 
-        [DllImport("Datadog.Trace.ClrProfiler.Native")]
+        [DllImport(DllName)]
         internal static extern PWArgs pw_createString(string s);
 
-        [DllImport("Datadog.Trace.ClrProfiler.Native")]
+        [DllImport(DllName)]
         internal static extern PWArgs pw_createInt(long value);
 
-        [DllImport("Datadog.Trace.ClrProfiler.Native")]
+        [DllImport(DllName)]
         internal static extern PWArgs pw_createUint(ulong value);
 
-        [DllImport("Datadog.Trace.ClrProfiler.Native")]
+        [DllImport(DllName)]
         internal static extern PWArgs pw_createArray();
 
-        [DllImport("Datadog.Trace.ClrProfiler.Native")]
+        [DllImport(DllName)]
         internal static extern PWArgs pw_createMap();
 
-        [DllImport("Datadog.Trace.ClrProfiler.Native")]
+        [DllImport(DllName)]
         internal static extern bool pw_addArray(ref PWArgs array, PWArgs entry);
 
         // Setting entryNameLength to 0 will result in the entryName length being re-computed with strlen
-        [DllImport("Datadog.Trace.ClrProfiler.Native")]
+        [DllImport(DllName)]
         internal static extern bool pw_addMap(ref PWArgs map, string entryName, ulong entryNameLength, PWArgs entry);
 
-        [DllImport("Datadog.Trace.ClrProfiler.Native")]
+        [DllImport(DllName)]
         internal static extern void pw_freeArg(ref PWArgs input);
 
 #pragma warning restore SA1300 // Element should begin with upper-case letter
+
+        private static List<string> GetDataDogNativeFolders(string runtimeId)
+        {
+            // first get anything "home folder" like, then combine with the profiler's location
+            // taking into account that this should be the same place
+
+            var homePaths = GetHomeFolders(runtimeId);
+            var profilerFolders = GetProfilerFolders();
+            var paths = homePaths.Concat(profilerFolders).Distinct().ToList();
+
+            return paths;
+        }
+
+        private static IEnumerable<string> GetProfilerFolders()
+        {
+            var profilerPathsEnvVars = new List<string>()
+            {
+                "CORECLR_PROFILER_PATH",
+                "CORECLR_PROFILER_PATH_32",
+                "CORECLR_PROFILER_PATH_64",
+                "COR_PROFILER_PATH",
```

Comment:
`COR_PROFILER_PATH_64` and `COR_PROFILER_PATH_32` are also supported on newer versions of .NET Framework. I could never find documentation of the exact version, but it was added around the 4.6 time frame.

Reply:
My idea had been simple to take any available environment variable, even if it wasn't were the process loaded the profiler from, we might find the library in the same location. However, this could create problems with loading the wrong version of the library, so I changed the method just to look at the env var that would be correct for the process type.